### PR TITLE
RI-6154 updates in cloud session

### PR DESCRIPTION
**Pull request by kchepikava**

### CODE REVIEW

File: redisinsight/api/migration/1729024196797-cloud-session.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class CloudSession1729024196797 implements MigrationInterface {
+    name = 'CloudSession1729024196797'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`CREATE TABLE "cloud_session" ("id" integer PRIMARY KEY NOT NULL, "data" varchar, "encryption" varchar)`);
```
Comment (ArtemHoruzhenko) [inline redisinsight/api/migration/1729024196797-cloud-session.ts:7]: let's make id to be a string type

File: redisinsight/api/src/modules/cloud/auth/cloud-auth.module.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { CloudAuthController } from 'src/modules/cloud/auth/cloud-auth.controlle
 import { CloudAuthAnalytics } from 'src/modules/cloud/auth/cloud-auth.analytics';
 
 @Module({
-  imports: [CloudSessionModule],
+  imports: [CloudSessionModule.register()],
```
Comment (ArtemHoruzhenko) [inline redisinsight/api/src/modules/cloud/auth/cloud-auth.module.ts:11]: no. we shouldn't call `register` here.

File: redisinsight/api/src/modules/cloud/capi-key/cloud-capi-key.module.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ import { CloudCapiKeyAnalytics } from 'src/modules/cloud/capi-key/cloud-capi-key
 @Module({
   imports: [
     CloudUserModule,
-    CloudSessionModule,
+    CloudSessionModule.register(),
```
Comment (ArtemHoruzhenko) [inline redisinsight/api/src/modules/cloud/capi-key/cloud-capi-key.module.ts:14]: lets avoid using `register` here too

File: redisinsight/api/src/modules/cloud/session/repositories/local.cloud.session.repository.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import { Repository } from 'typeorm';
+import { classToClass } from 'src/utils';
+// import { SessionMetadata } from 'src/common/models';
+import { EncryptionService } from 'src/modules/encryption/encryption.service';
+import { ModelEncryptor } from 'src/modules/encryption/model.encryptor';
+import { CloudSessionRepository } from './cloud.session.repository';
+import { CloudSessionEntity } from '../entities/cloud.session.entity';
+import { CloudSessionData } from '../models/cloud-session';
+
+const SESSION_ID = 1;
```
Comment (ArtemHoruzhenko) [inline redisinsight/api/src/modules/cloud/session/repositories/local.cloud.session.repository.ts:11]: let's make it a string type

File: redisinsight/api/src/modules/cloud/session/repositories/local.cloud.session.repository.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      return null;
+    }
+
+    const decrypted = await this.modelEncryptor.decryptEntity(entity, false);
+    return classToClass(CloudSessionData, decrypted);
+  }
+
+  async save(cloudAuth: Partial<CloudSessionData>): Promise<void> {
+    const entity = await this.modelEncryptor.encryptEntity(
+      classToClass(CloudSessionEntity, { id: SESSION_ID, ...cloudAuth }),
```
Comment (ArtemHoruzhenko) [inline redisinsight/api/src/modules/cloud/session/repositories/local.cloud.session.repository.ts:38]: I would like to rewrite this to `{  ...cloudAuth, id: SESSION_ID }` to make sure that id will be hardcoded

File: redisinsight/api/src/modules/cloud/subscription/cloud-subscription.module.ts [hunk 6]
```diff
@@ -11,7 +11,7 @@ import { CloudSubscriptionCapiProvider } from './providers/cloud-subscription.ca
 @Module({
   imports: [
     CloudUserModule,
-    CloudSessionModule,
+    CloudSessionModule.register(),
```
Comment (ArtemHoruzhenko) [inline redisinsight/api/src/modules/cloud/subscription/cloud-subscription.module.ts:14]: we must avoid it

File: redisinsight/api/src/modules/cloud/user/cloud-user.module.ts [hunk 7]
```diff
@@ -11,7 +11,7 @@ import { CloudAuthModule } from 'src/modules/cloud/auth/cloud-auth.module';
 
 @Global()
 @Module({
-  imports: [CloudSessionModule, CloudAuthModule],
+  imports: [CloudSessionModule.register(), CloudAuthModule],
```
Comment (ArtemHoruzhenko) [inline redisinsight/api/src/modules/cloud/user/cloud-user.module.ts:14]: and here

**Review by ArtemHoruzhenko — APPROVED** [review]

**Review by mariasergeenko — APPROVED** [review]

**Review by mariasergeenko — APPROVED** [review]

**Pull request merged**